src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -174,6 +183,46 @@ protected virtual void BeginInvokeJS(long taskId, string identifier, string? arg
             DotNetInvocationInfo invocationInfo,
             in DotNetInvocationResult invocationResult);
 
+        /// <summary>
+        /// Invoked by JS
+        /// </summary>
+        /// <param name="id">Atomically incrementing identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered to JS.</param>
+        protected internal virtual void SendByteArray(int id, byte[] data)
+        {
+            throw new NotSupportedException("JSRuntime subclasses are responsible for implementing byte array transfer to JS.");
+        }
+
+        /// <summary>
+        /// Accepts the byte array data being transferred from JS to DotNet.
+        /// </summary>
+        /// <param name="id">Identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered from JS.</param>
+        public void ReceiveByteArray(int id, byte[] data)
```

Comment:
I think we need to have this be `virtual` so that RemoteJSRuntime can be the one doing the check about the MessageSize. We don't want to impose that limitation in other places where this doesn't matter, like WebAssembly.